process/factory/shard/vmContainerFactory.go

```diff
@@ -104,15 +108,12 @@ func (vmf *vmContainerFactory) Create() (process.VirtualMachinesContainer, error
 
 // EpochConfirmed updates VMs in all created containers depending on the epoch
 func (vmf *vmContainerFactory) EpochConfirmed(epoch uint32) {
-	if epoch >= vmf.ArwenGasManagementRewriteEnableEpoch {
-		for _, container := range vmf.containers {
-			newArwenVM, err := vmf.createInProcessArwenVMByEpoch(epoch)
-			if err != nil {
-				logVMContainerFactory.Error("cannot replace Arwen VM in epoch", "epoch", epoch)
-				continue
-			}
-			container.Replace(factory.ArwenVirtualMachine, newArwenVM)
-		}
+	if vmf.config.OutOfProcessEnabled {
+		return
+	}
+
+	if epoch == vmf.ArwenGasManagementRewriteEnableEpoch {
```

Comment:
I think we need an array here: to specify all epochs when the arwen binary should be changed.

Reply:
do we want to integrate with vector of epochs already ? This condition is not enough as there are edgecases when you go back to the previous epoch.
Thus you need something more complex - like currentRunningVersion and if epoch >= arwenV4 -> replaceArwen with newArwen. Like you have to verify the current running version vs what you suppose to run and replace according to it.